site/src/components/introduction/building-a-chart.md

```diff
@@ -0,0 +1,188 @@
+---
+layout: component
+namespace: Introduction
+title: Building A Chart
+externals:
+  building-a-chart-complete-js: building-a-chart-complete.js
+  building-a-chart-complete-html: building-a-chart-complete.html
+  building-a-chart-complete-css: building-a-chart-complete.css
+  creating-some-data-html: creating-some-data.html
+  creating-some-data-js: creating-some-data.js
+  rendering-a-series-html: rendering-a-series.html
+  rendering-a-series-js: rendering-a-series.js
+  cartesian-chart-html: cartesian-chart.html
+  cartesian-chart-js: cartesian-chart.js
+  cartesian-chart-extent-html: cartesian-chart-extent.html
+  cartesian-chart-extent-js: cartesian-chart-extent.js
+  tidy-chart-html: tidy-chart.html
+  tidy-chart-js: tidy-chart.js
+  adding-annotations-html: adding-annotations.html
+  adding-annotations-js: adding-annotations.js
+---
+
+This brief tutorial introduces the core concepts behind d3fc, via the process of building a chart. You'll find out why we created d3fc and what makes it different from the numerous other charting libraries out there!
+
+Here's the chart you're going to be building:
+
+<style>
+{{{building-a-chart-complete-css}}}
+</style>
+
+{{{ dynamic-include 'codepen' html="building-a-chart-complete-html" js="building-a-chart-complete-js"}}}
+{{{ building-a-chart-complete-html }}}
+
+It's a pretty conventional 'sales dashboard' chart, in this case showing some fictional monthly sales data, with a couple of lines indicating targets.
+
+But before we start putting this chart together, let's look at charting libraries in a more general sense ...
+
+## Charting APIs
+
+Charts are complex UI components, they are designed to be highly flexible in order to support a wide range of different chart types, styles and configurations. However, this flexibility is most often presented as a complex API of hundreds of classes and thousands of methods. Searching for the correct configuration that will yield the result you are after is a challenge, and you are heavily reliant on the designer of the chart having considered your use-case to make that configuration possible.
+
+D3 offers a fresh perspective, offering a different way to create charts and visualisations. It is an expressive, yet low-level tool, that if wielded correctly, can be used to create practically any visual representation you like.
+
+There are a number of charting libraries built with D3, however these invariably hide the underlying power of D3 behind the same-old complex and confusing APIs.
+
+With d3fc our aim is to enhance D3 by providing a suite of powerful charting components that make it easy for you to assemble exactly the chart you want. These components 'speak the language' of D3, supporting selections and datajoins. Furthermore, using the decorate pattern, you can enhance and manipulate the way in which they are rendered.
```

Comment:
As mentioned, there are quite a few D3 terms here - is it worth providing some useful links?

E.g.

- charting components - http://bost.ocks.org/mike/chart/
- selections - http://bost.ocks.org/mike/selection/
- data joints - http://bost.ocks.org/mike/join/
- decorate pattern - https://d3fc.io/components/introduction/decorate-pattern.html